don't die if we happen to use GNU arch instead of Darwin arch

http://bugs.python.org/issue7715

--- configure.in
+++ configure.in
@@ -1593,10 +1593,10 @@
     
     if test "${ac_osx_32bit}" = "yes"; then
     	case `arch` in
-    	i386) 
+    	i386|x86_64) 
     		MACOSX_DEFAULT_ARCH="i386" 
     		;;
-    	ppc) 
+    	ppc|"Power Macintosh") 
     		MACOSX_DEFAULT_ARCH="ppc" 
     		;;
     	*)
@@ -1605,10 +1605,10 @@
     	esac
     else
     	case `arch` in
-    	i386) 
+    	i386|x86_64) 
     		MACOSX_DEFAULT_ARCH="x86_64" 
     		;;
-    	ppc) 
+    	ppc|"Power Macintosh") 
     		MACOSX_DEFAULT_ARCH="ppc64" 
     		;;
     	*)
